Code/Framework/AzCore/AzCore/Component/EntitySerializer.h

```diff
@@ -30,4 +30,23 @@ namespace AZ
         void ConvertComponentVectorToMap(const AZ::Entity::ComponentArrayType& components,
             AZStd::unordered_map<AZStd::string, AZ::Component*>& componentMapOut);
     };
+
+    /**
+     * Helper class to track components that have been skipped during loading.
+     * When this class is added to the metadata of a Json deserializer setting,
+     * custom component serializers can add themselves to the list so users
+     * can be informed of component deprecation upon load completion
+     */
+    class DeprecatedComponentMetadata
+    {
+    public:
+        AZ_RTTI(DeprecatedComponentMetadata, "{3D5F5EAE-BDA9-43AA-958E-E87158BAFB9F}");
+        virtual ~DeprecatedComponentMetadata() {}
```

Comment:
nit: It might be better to have this be `= default` instead of having the empty braces

Reply:
Also, it doesn't look like this class needs to have a `virtual` destructor.

Reply:
I think it does unfortunately because of the AZ_RTTI macro 🙁 This type might actually just need AZ_TYPE_INFO thinking about it if nothing is going to inherit from this and need azdynamic_cast. If we do that we can then avoid the virtual destructor 👍 🥳

Reply:
Oh yeah, copy/paste error. Thanks, I'll address the feedback and post an update.

Reply:
Turns out it needs the virtual destructor due to the AZ_RTTI macro.

Reply:
The `AZ_TYPE_INFO` macro can be used instead.

Reply:
+1 (I did mention that, just in a long-winded way above 😅)